File: Blackjack/Card.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Net.NetworkInformation;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using static System.Reflection.Metadata.BlobBuilder;

namespace Blackjack2
{
    internal class Card
    {
        public enum Suit { Hearts = 1, Diamonds, Clubs, Spades };
        public enum Rank { Ace = 1, Two, Three, Four, Five, Six, Seven, Eight, Nine, Ten, Jack, Queen, King };
        public Suit suit;
        public Rank rank;


        public Card()
        {
        	rank = Rank.Ace;		//default original rank
	        suit = Suit.Hearts;	    //default original suit
        }

    public void Display_Card()   //displays a card in text
    {
            Console.Write($"{rank} of {suit}");
    }

    public void Set_Suit(int x)      //sets a card's suit
    {
        if (x == 1)
            suit = Suit.Hearts;
        else if (x == 2)
            suit = Suit.Diamonds;
        else if (x == 3)
            suit = Suit.Clubs;
        else if (x == 4)
            suit = Suit.Spades;
        else
            Console.Write("Error");
    }

    public void Set_Rank(int x)      //sets a card's rank
    {
        rank = (Rank)x;
    }

    public int Get_Value(bool flag)     //returns the value of a card
    {
        if (rank == Rank.Ace)
            if (flag == true)           //flag is true if total is over 21, and an Ace must be counted as a 1
            {
                return 1;
                flag = false;           //reset flag, to accommodate multiple aces in one hand
            }
            else
                return 11;              //by default, aces are counted as 11 originally
        else if (rank == Rank.Two)
            return 2;
        else if (rank == Rank.Three)
            return 3;
        else if (rank == Rank.Four)
            return 4;
        else if (rank == Rank.Five)
            return 5;
        else if (rank == Rank.Six)
            return 6;
        else if (rank == Rank.Seven)
            return 7;
        else if (rank == Rank.Eight)
            return 8;
        else if (rank == Rank.Nine)
            return 9;
        else if (rank == Rank.Ten)
            return 10;
        else if (rank == Rank.Jack)
            return 10;
        else if (rank == Rank.Queen)
            return 10;
        else if (rank == Rank.King)
            return 10;
        else
            return 0;
    }

        public string Get_DisplayRank()     //returns the display value of a card
        {
            if (rank == Rank.Ace)
                return "A ";
            else if (rank == Rank.Two)
                return "2 ";
            else if (rank == Rank.Three)
                return "3 ";
            else if (rank == Rank.Four)
                return "4 ";
            else if (rank == Rank.Five)
                return "5 ";
            else if (rank == Rank.Six)
                return "6 ";
            else if (rank == Rank.Seven)
                return "7 ";
            else if (rank == Rank.Eight)
                return "8 ";
            else if (rank == Rank.Nine)
                return "9 ";
            else if (rank == Rank.Ten)
                return "10";
            else if (rank == Rank.Jack)
                return "J ";
            else if (rank == Rank.Queen)
                return "Q ";
            else if (rank == Rank.King)
                return "K ";
            else
                return "  ";
        }
        public char Get_DisplaySuit()
        {
            if (suit == Suit.Hearts)
                return 'H';
            if (suit == Suit.Diamonds)
                return 'D';
            if (suit == Suit.Clubs)
                return 'C';
            if (suit == Suit.Spades)
                return 'S';
            else
                return 'X';
        }

        public void Draw_Back()              //draws the back of a card (for hold cards)
        {
            Console.Write($"\t┌───────────┐\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t│***********│\n");
            Console.Write($"\t└───────────┘\n");

            Console.Write("\tHold Card");
            Console.WriteLine("\n");
        }

        public void Draw_Card()          //draws a specific card's picture
        {

            Console.Write($"\t┌───────────┐\n");
            Console.Write($"\t│ {Get_DisplayRank()}        │\n");
            Console.Write($"\t│           │\n");
            Console.Write($"\t│           │\n");
            Console.Write($"\t│           │\n");
            Console.Write($"\t│     {Get_DisplaySuit()}     │\n");
            Console.Write($"\t│           │\n");
            Console.Write($"\t│           │\n");
            Console.Write($"\t│           │\n");
            Console.Write($"\t│        {Get_DisplayRank()} │\n");
            Console.Write($"\t└───────────┘\n");

            Console.Write("\t");
            Display_Card();
            Console.WriteLine("\n");
        }
    }
}

File: PythonBlackjack/Card.py
Suit = [
    "HEARTS",
    "DIAMONDS",
    "CLUBS",
    "SPADES"
]

Rank = [
    "ACE",
    "TWO",
    "THREE",
    "FOUR",
    "FIVE",
    "SIX",
    "SEVEN",
    "EIGHT",
    "NINE",
    "TEN",
    "JACK",
    "QUEEN",
    "KING"
]
class Card:
    @property
    def suit(self):
        return self.__suit

    @property
    def rank(self):
        return self.__rank

    def __init__(self, suit=None, rank=None):
        if suit is None:
            suit = 0  # default to hearts
        if rank is None:
            rank = 0  # default to ACE (index 0)
        self.__set_suit(suit)
        self.__set_rank(rank)

    def get_suit_index(self, value):
        if isinstance(value, str):
            return Suit.index(value.upper())
        elif isinstance(value, int) and value >= 0 and value < 4:
            return value
        else:
            raise ValueError("Suit must be an integer between 1 and 4 or a valid suit name")

    def get_rank_index(self, value):
        if isinstance(value, str):
            return Rank.index(value.upper())
        elif isinstance(value, int) and 0 <= value < 13:
            return value
        else:
            raise ValueError("Rank must be an integer between 0 and 12 or a valid rank name")

    def __set_suit(self, x):  # sets a card's suit
        self.__suit = self.get_suit_index(x)

    def __set_rank(self, x):
        self.__rank = self.get_rank_index(x)

    def display_card_as_text(self):  # displays a card in text
        print(Rank[self.__rank] + " of " + Suit[self.__suit])

    def get_value(self, flag):  # returns the value of a card
        # Returns the value of a card
        if self.rank == 0:  # ACE
            return 1 if flag else 11
        elif 1 <= self.rank <= 8:  # TWO to NINE
            return self.rank + 1
        elif 9 <= self.rank <= 12:  # TEN, JACK, QUEEN, KING
            return 10
        else:
            return 0

    def get_display_rank(self, short=False):  # returns the display value of a card
        if self.rank == 0:
            return "A" if short else "A "
        elif 1 <= self.rank <= 8:
            return str(self.rank + 1) if short else str(self.rank + 1) + " "
        elif self.rank == 9:
            return "T" if short else "10"
        elif self.rank == 10:
            return "J" if short else "J "
        elif self.rank == 11:
            return "Q" if short else "Q "
        elif self.rank == 12:
            return "K" if short else "K "
        else:
            return " " if short else "  "

    def get_display_suit(self):
        if self.suit == Suit.index("HEARTS"):
            return '\u2665'
        if self.suit == Suit.index("DIAMONDS"):
            return '\u2666'
        if self.suit == Suit.index("CLUBS"):
            return '\u2663'
        if self.suit == Suit.index("SPADES"):
            return '\u2660'
        else:
            return ' '